lib/relation-definition.js

```diff
@@ -1596,17 +1638,17 @@ RelationDefinition.hasAndBelongsToMany = function hasAndBelongsToMany(modelFrom,
  * has only one account.
  *
  * @param {Function} modelFrom The declaring model class
- * @param {String|Function} modelTo Model object (or String name of model) to
- * which you are creating the relationship.
+ * @param {Object|String} modelToRef Reference to Model object to which you are
+ *  creating the relation: model instance, model name, or name of relation to model.
  * @options {Object} params Configuration parameters; see below.
  * @property {String} as Name of the property in the referring model that
  * corresponds to the foreign key field in the related model.
  * @property {String} foreignKey Property name of foreign key field.
  * @property {Object} model Model object
  */
-RelationDefinition.hasOne = function(modelFrom, modelTo, params) {
+RelationDefinition.hasOne = function(modelFrom, modelToRef, params) {
   params = params || {};
-  modelTo = lookupModelTo(modelFrom, modelTo, params);
+  var modelTo = lookupModelTo(modelFrom, modelToRef, params);
 
   var pk = params.primaryKey || modelFrom.dataSource.idName(modelFrom.modelName) || 'id';
   var relationName = params.as || i8n.camelize(modelTo.modelName, true);
```

Comment:
same here: `params.reference || params.as || i8n.camelize(modelTo.pluralModelName, true);` if we decide to use reference